### PR TITLE
SOLR-17576: Remove deprecated master/slave support in ReplicationHandler

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-17576

# Description

Remove deprecated terms.

# Solution

Deletes

# Tests

Existing tests

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended, not available for branches on forks living under an organisation)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by gus-asf** [comment]
I'm confused. There seems to be a lot more in this than just touching up terminology to match the Leader/Follower?

File: solr/core/src/java/org/apache/solr/handler/IndexFetcher.java [hunk 1]
```diff
@@ -211,7 +205,7 @@ public static class IndexFetchResult {
         new IndexFetchResult("Local index commit is already in sync with peer", true, null);
 
     public static final IndexFetchResult INDEX_FETCH_FAILURE =
-        new IndexFetchResult("Fetching lastest index is failed", false, null);
+        new IndexFetchResult("Fetching latest index is failed", false, null);
```
Comment (gus-asf) [inline solr/core/src/java/org/apache/solr/handler/IndexFetcher.java:208]: "Fetching the latest index failed" ?

File: solr/core/src/java/org/apache/solr/handler/IndexFetcher.java [hunk 2]
```diff
@@ -1526,49 +1512,6 @@ private void copyTmpConfFiles2Conf(Path tmpconfDir) {
     }
   }
 
-  /**
-   * The tlog files are moved from the tmp dir to the tlog dir as an atomic filesystem operation. A
-   * backup of the old directory is maintained. If the directory move fails, it will try to revert
-   * back the original tlog directory.
-   */
-  private boolean copyTmpTlogFiles2Tlog(File tmpTlogDir) {
```
Comment (gus-asf) [inline solr/core/src/java/org/apache/solr/handler/IndexFetcher.java:1534]: how is this related to this ticket?

File: solr/core/src/java/org/apache/solr/handler/IndexFetcher.java [hunk 3]
```diff
@@ -1611,23 +1554,6 @@ private Collection<Map<String, Object>> getModifiedConfFiles(
     return nameVsFile.isEmpty() ? Collections.emptyList() : nameVsFile.values();
   }
 
-  /**
```
Comment (gus-asf) [inline solr/core/src/java/org/apache/solr/handler/IndexFetcher.java:1614]: How is this related?

File: solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java [hunk 4]
```diff
@@ -848,14 +802,6 @@ private Date getNextScheduledExecTime() {
     return nextTime;
   }
 
-  int getTimesReplicatedSinceStartup() {
```
Comment (gus-asf) [inline solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java:851]: how is this change related to the ticket?

File: solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java [hunk 5]
```diff
@@ -1577,28 +1522,16 @@ private Long readIntervalNs(String interval) {
...
-   * @deprecated Only used for backwards compatibility. Use {@link #LEADER_URL}
-   */
-  @Deprecated public static final String LEGACY_LEADER_URL = "masterUrl";
-
   public static final String FETCH_FROM_LEADER = "fetchFromLeader";
 
   // in case of TLOG replica, if leaderVersion = zero, don't do commit
-  // otherwise updates from current tlog won't copied over properly to the new tlog, leading to data
+  // otherwise updates from current tlog won't be copied over properly to the new tlog, leading to
+  // data
```
Comment (gus-asf) [inline solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java:1529]: tidy not really helping here... :)

File: solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java [hunk 6]
```diff
@@ -1637,8 +1570,6 @@ private Long readIntervalNs(String interval) {
 
   public static final String ALIAS = "alias";
 
-  public static final String CONF_CHECKSUM = "confchecksum";
```
Comment (gus-asf) [inline solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java:1640]: unrelated?

**Review by gus-asf — CHANGES_REQUESTED** [review]
some of the trivial stuff like spelling fixes isn't of concern, but I assume some of the larger deletions are removals of unused methods... I think that type of thing should be it's own change so that the reason for removal can be clearer to anyone who might have used it in 3rd party code or forks of our repo...

**Comment by epugh** [comment]
@gus-asf unfortuantly, I have to agree with you....   I'm going to redo this PR to seperate out the code cleanups from the deprecations.   I just read @dsmiley link about "how to maintain forks" and it really pushed the atomicity of PR's.    It kills me to be going through code and seeing obvious errors and problems and NOT fix them.   The next time, I might try having two copies of Solr in seperate directories, and try to keep "annonying code issue fixes" on one, and the thrust of whatever I am doing on the other....

**Comment by epugh** [comment]
I've removed the extra refactorings and created a new issue https://issues.apache.org/jira/browse/SOLR-17579 to track that (and remind me!)
